torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1041,9 +1043,166 @@ def track_symint(source, val):
+
+            def __sub__(self, other):
+                return myz3Ref(super().__sub__(other))
+
+            def __rsub__(self, other):
+                return myz3Ref(super().__rsub__(other))
+
+            def __mul__(self, other):
+                return myz3Ref(super().__mul__(other))
+
+            def __rmul__(self, other):
+                return myz3Ref(super().__rmul__(other))
+
+            def __floor__(self):
+                breakpoint()
+                return myz3Ref(super().__floor__())
+
+            def __floordiv__(self, other):
+                div = self / other
+                return myz3Ref(div if div.is_int() else z3.ToInt(div))
+
+            def __rfloordiv__(self, other):
+                div = other / self
+                return myz3Ref(div if div.is_int() else z3.ToInt(div))
+
+            def __pow__(self, other):
+                if isinstance(other, int):
+                    return myz3Ref(functools.reduce(operator.mul, [self] * other))
+                return myz3Ref(super().__pow__(other))
+
+        def myz3Int(name):
+            return myz3Ref(z3.Int(name))
+
+        class Z3SymbolicTensor:
+            def __init__(self, name, max_idx):
+                self._size   = [myz3Int(f"{name}.size()[{i}]") for i in range(max_idx+1)]
+                self._stride = [myz3Int(f"{name}.stride()[{i}]") for i in range(max_idx+1)]
+                self._storage_offset = myz3Int(f"{name}.storage_offset()")
+
+            def size(self):
+                return self._size
+
+            def stride(self):
+                return self._stride
+
+            def storage_offset(self):
+                return self._storage_offset
+
+
+        vars = {}
+        # breakpoint()
+        for vs in symbol_to_source.values():
+            for v in vs:
+                name = v.base.name()
+                vars[name] = max(vars.get(name, 0), v.idx or 0)
+
+        for source, expr in input_guards:
+            name = source.base.name()
+            vars[name] = max(vars.get(name, 0), source.idx or 0)
+
+        remapped_names = {}
+        for name, idx in vars.items():
+            # a name can be an array access, so if it has `[` `]` in it, we have
+            # to pase that out and nest the structure accordingly
+            # name = name.replace(".", "DYNAMODOTSWAPSLUG")
+            if name in remapped_names:
+                continue
+
+            def potential_rename(name):
+                if "." in name:
+                    # NNModule shenanigins?
+                    real_value = eval(name, scope)
+                    # assert isinstance(real_value, torch.nn.Module), breakpoint()
+                    new_name = name.replace(".", "_")
+                    remapped_names[name] = new_name
+                    # name = new_name
+                    # breakpoint()
+                    return new_name
+                return name
+
+            if "[" in name:
+                assert "]" in name
```

Comment:
this 100% doesn't work for something like `[k][n]...`, but have not yet encountered it, still , easy to support and make `[k]` work within it